### PR TITLE
Splitting `pulser` into `pulser-core` and `pulser-simulation`.

### LINKED ISSUE #358: Splitting the `pulser.simulation` module
## Objectives:
- Have all the features of the `pulser.simulation` module in a separate distribution, thus allowing for a leaner `pulser-core` distribution that's entirely focused on the sequence design aspect.
- Keep existing code working, at least until the next major release (which will be 1.0.0)
- Minimize the need for code adaptations on the user side as much as possible
- `pip install pulser` would still install everything, but now there is the option to `pip install pulser-core`(or similar) for situations where the simulation package is not needed.

### 1. Namespace packages

[Namespace packages](https://packaging.python.org/en/latest/guides/packaging-namespace-packages/) allow for a single package to be split into multiple sub-packages, which can be distributed, installed and used independently.

The advantage of this approach is that we could make `pulser` a namespace package and `pulser-simulation` a subpackage, which would allow us to keep the `from pulser.simulation import ...` imports unchanged. It would also mean that all the other modules would be moved to a `pulser-core` subpackage. We would thus move from this:

```
pulser/
    __init__.py
    setup.py
    simulation/
        __init__.py
        ...
    devices/
        __init__.py
        ...
    sequence.py
    [all the other pulser modules]
```

to this:

```
pulser-simulation/
    setup.py
    pulser/
        simulation/
            __init__.py
            ...

pulser-core/
    setup.py
    pulser/
        devices/
            __init__.py
            ...
        sequence.py
        [all the other pulser modules]
```

When `pulser-simulation` and `pulser-core` are installed, their contents would then be located under the `pulser` namespace package.

As far as I could tell, not being able to have an `__init__.py` means that we have to drop the direct imports from `pulser` (like `from pulser import Sequence`), which would defeat the purpose of keeping existing code working. I haven't experiment with this yet, so perhaps there is a way, but if there is it is obscure. **If there is indeed no way to do this, I would say this makes this approach unviable.**

### 2. Make a separate `pulser_simulation` package

This is a more straightforward solution, which would essentially just extract the simulation module and keep the rest the same. In this case, the file structure would look something like this.

```
pulser-simulation/
    setup.py   # pulser_simulation package setup, with pulser as a dependency (distribution: 'pulser-simulation')
    pulser_simulation/
        __init__.py
        simulation.py
        simresults.py
        ...         

pulser-core/
    setup.py   # pulser package setup (distribution: 'pulser-core')
    pulser/
        devices/
            __init__.py
            ...
        sequence.py
        [all the other pulser modules]
setup.py  # 'pulser' distribution, just installs 'pulser-core' and 'pulser-simulation'
```

As it is, this approach will have the inconvenience of requiring that all imports of `pulser.simulation` be changed to `pulser_simulation`. I think a workaround here is to keep an empty `simulation` module inside `pulser` with only an `__init__.py` containing code that checks if `pulser_simulation` is installed and imports the relevant classes from it (i.e. `Simulation` and `SimConfig`). I think this would be fine for now, I'm not so sure it's something we would like to keep forever or deprecate at some point.

### PR DESCRIPTION
**Pull request by HGSilveri**
## Distributions

Following the motivations and objectives in #358, this PR splits the current `pulser` distribution into two:

- `pulser-simulation`, which installs the `pulser_simulation` package and contains all contents previously in `pulser.simulation`
- `pulser-core`, which installs the `pulser` package containing everything else.

Note that the `pulser` distribution is still kept, but it becames only a meta-package that installs `pulser-core` and `pulser-simulation`.

## Versions

In terms of versioning, I went with a global version for all distributions and packages, which is recorded in `VERSION.txt`.

## Backwards compatibility

Support for importing `Simulation` and `SimConfig` from `pulser.simulation` is kept (if `pulser-simulation` is installed), but a warning is raised recommending imports are switched to `pulser_simulation`. **I'm still considering if we should update all imports on the tutorials right away or not**.

- [ ] Consider updating the `pulser.simulation` imports on the tutorials

## Advice for reviewers

Almost all files were touched, though many of them were only renamed. The commits where files are moved purposefully do not introduce any changes, so it is advised that the changes are reviewed in a commit-by-commit basis, such that the file renames don't pollute the diff.

Closes #358 .

### CODE REVIEW

File: pulser-core/README.md
Comment (bejito) [inline pulser-core/README.md:1]: nit: pulser-core

File: pulser-core/rtd_requirements.txt
Comment (bejito) [inline pulser-core/rtd_requirements.txt:1]: What is this for ?

**Review by bejito — APPROVED** [review]
Minor comment, otherwise LGTM

File: pulser-core/rtd_requirements.txt
Comment (HGSilveri) [reply pulser-core/rtd_requirements.txt:1]: This is needed for the Read The Docs builder to install `pulser-core`

File: pulser-core/README.md
Comment (HGSilveri) [reply pulser-core/README.md:1]: Nice catch, thanks!

**Comment by HGSilveri** [comment]
Thanks @bejito ! Do you think this will have an impact on the work being done on serialization to the abstract representation, or is it okay to merge right away?

**Review by sebgrijalva — APPROVED** [review]
Looks good! Thanks for the work here.

**Comment by bejito** [comment]
There shouldn't be any impact on serialization, I think it is safe to merge.

File: pulser-simulation/setup.py
Comment (HGSilveri) [reply pulser-simulation/setup.py:41]: I would say we leave it to the `pulser-core` installation, unless we have version requirements particular to `pulser-simulation` on those packages, in which case we specify them here.

File: README.md
Comment (HGSilveri) [reply README.md:18]: Yes, thanks!

**Pull request merged**
